### PR TITLE
8180571: Refactor sun/security/pkcs11 shell tests to plain java tests and fix failures

### PR DESCRIPTION
**Pull request by fguallini**
Refactor the following shell tests to Java:
- security/pkcs11/KeyStore/Basic.sh
- security/pkcs11/KeyStore/ClientAuth.sh
- security/pkcs11/KeyStore/SecretKeysBasic.sh
- security/pkcs11/Provider/ConfigQuotedString.sh
- security/pkcs11/Provider/Login.sh
- security/pkcs11/Config/ReadConfInUTF16Env.sh

Currently, most of the shell tests in the list may be ignored during execution time in most platforms since they are incorrectly filtered out by the OS name they are run on. For example, ClientAuth.sh is only run if the OS name is equal to ‘Linux’, but OS name may also include the architecture such as ‘Linux x86_64’. Those platform constraints are removed in this PR.

Additionally, further changes are introduced in the following test:

- ClientAuth: it was failing intermittently because the server side was binding to the wildcard address. The issue is fixed by binding to loopback address instead. Also, Thread.sleep is replaced with CountdownLatch to facilitate synchronization between client and server. Finally, a new ‘user1’ certificate was generated since the current one has expired.

- Basic: Remove redundant X509Certificate casting 

- SecretKeysBasic: it was already in the problem list since it reproduces the open bug JDK-8209398 and fails. Test is refactored to java and still reproduces the issue.

All the mentioned tests were run many times in multiple platforms to ensure stability

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8180571](https://bugs.openjdk.java.net/browse/JDK-8180571): Refactor sun/security/pkcs11 shell tests to plain java tests and fix failures

### Reviewers
 * [Xue-Lei Andrew Fan](https://openjdk.java.net/census#xuelei) (@XueleiFan - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4092/head:pull/4092` \
`$ git checkout pull/4092`

Update a local copy of the PR: \
`$ git checkout pull/4092` \
`$ git pull https://git.openjdk.java.net/jdk pull/4092/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4092`

View PR using the GUI difftool: \
`$ git pr show -t 4092`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4092.diff">https://git.openjdk.java.net/jdk/pull/4092.diff</a>

</details>

### CODE REVIEW

**Comment by fguallini** [comment]
Please if someone could review and sponsor this PR. Thanks

File: test/jdk/sun/security/pkcs11/Config/ReadConfInUTF16Env.java [hunk 1]
```diff
@@ -21,17 +21,36 @@
  * questions.
  */
 
+/* @test
+ * @bug 8187023
```
Comment (sisahoo) [inline test/jdk/sun/security/pkcs11/Config/ReadConfInUTF16Env.java:25]: Should it have the enhancement bug id too.

Comment (fguallini) [reply test/jdk/sun/security/pkcs11/Config/ReadConfInUTF16Env.java:25]: My understanding is that we should only include the id of product bugs, no test enhancements. That was a comment somebody added in a previous PR. The reason is that you can track changes in the test files with git history, whereas it is only possible to track the related product bug id by adding it manually to the class header

**Comment by fguallini** [comment]
Please if someone could review and sponsor this PR. Thanks

**Review by XueleiFan — APPROVED** [review]
Nice update.  Look good to me.

**Pull request closed without merging**